main.py

```diff
@@ -153,6 +184,7 @@ def _add_new_stream(self, stream_url=None, stream_quality=cfg[CONFIG_QUALITY]):
                 self.fail_add_stream.emit(AddStreamError.DEFAULT_QUALITY_MISSING, (stream_options, stream_url, stream_quality))
                 return
 
+            time.sleep(delay)
```

Comment:
Could we do this only if a delay was specified? Right now it would call time.sleep for every stream (although with the value 0 for some) but we could instead ignore the sleep if the value is something other than 0. Might not make a difference in practice but just something I thought of. It might schedule a sleep for a period of 0, but the scheduling itself might be a overhead. I have no idea. What do you think?